Singleton/src/Singleton.cpp

```diff
@@ -0,0 +1,23 @@
+#include "Singleton.hpp"
+
+// Singleton factory
+Example* SingletonFactory::get() {
+    static Example e{};
+    return &e;
+};
+
+// Not a singleton factory
+Example* NotSingletonFactory::get() {
+    return new Example{};
+};
+
+// Singleton Tester
+template <typename T>
```

Comment:
The definition of a template function should NOT be in the .cpp file as the template classes cannot be compiled.
When you are having a template class MyClass<T> and you want to instantiate two objects in your code like so:
`MyClass<int> obj1;
MyClass<std::string> obj2;`
the compiler will generate two different classes. Think of the compiler as an oven and the classes are cookies. In this case, the template is just a cookie cutter. You don't put the cookie cutter in the oven unless you want an explosion, you just use it to model the shape of the cookies.

Reply:
I've seen a lot of tutorial that, when approaching templates (classes or functions) they said the same thing but because they receive an error if they do not respect this. I've tried this myself and did not get an error so I thought that it's alright.
Ok then, I'll move the definition in the hpp file.